### PR TITLE
Reimplement the :firstn strategy

### LINKED ISSUE #30: Reintroduce `pinthreads(:firstn)`
Doing essentially `pinthreads(cpuids_all()[1:nthreads()])`.

### PR DESCRIPTION
**Pull request by JamesWrigley**
Fixes #30.

### CODE REVIEW

File: src/pinning.jl [hunk 1]
```diff
@@ -137,6 +140,7 @@ Per default, a reasonable value for `places` is chosen based on the given `pinni
 * `:spread` or `scatter`: pins to `places` in an alternating / round robin fashion.
 * `:random`: shuffles the given `places` and then pins to them compactly.
 * `:current`: pins threads to the cpu threads where they are currently running (ignores `places`).
+* `:firstn`: pin threads to the cpu ids `0:nthreads - 1`.
```
Comment (carstenbauer) [inline src/pinning.jl:143]: This is only correct if the first n cpu ids are indeed 0,1,2,3,... etc. which isn't the case on all systems. Instead, `:firstn` pins to the first n cpuids (as reported by lscpu, which is the basis for everything in ThreadPinning.jl).

File: src/pinning.jl [hunk 2]
```diff
@@ -125,6 +127,7 @@ function getcpuids_pinning(::RandomBind, places; kwargs...)
     reduce(vcat, Random.shuffle(getplaces_cpuids(places)))
 end
 getcpuids_pinning(::CurrentBind, args...; kwargs...) = getcpuids()
+getcpuids_pinning(::FirstNBind, args...; kwargs...) = cpuids_all()[1:nthreads()]
```
Comment (carstenbauer) [inline src/pinning.jl:130]: Might need a bounds check. For example, `nthreads()` could be larger than the number of available cpu ids.

**Review by carstenbauer — COMMENTED** [review]
Thanks for the PR! See my comments :)

File: src/pinning.jl (hunk 1 above)
Comment (JamesWrigley) [reply src/pinning.jl]: Right, I keep forgetting that :facepalm: Fixed in e9b3810.

**Comment by JamesWrigley** [comment]
BTW Github let's you explicitly approve a PR, if you do that or comment your approval, I'll squash the fixup comments before merging :) I meant to do that for #17.

**Review by carstenbauer — APPROVED** [review]

**Pull request merged**